### ISSUE TITLE
Failing to expose my server / receiving campaigns statistics

### ISSUE THREAD

**Issue by JulesClaussen** [gophish/gophish#2636]
Hello everyone!
I'm struggling a bit to collect statistics (email opened only for the moment!) for my gophish campaigns.

What version of Gophish are you using?: 
0.12.1

Brief description of the issue: 
I am running locally (MacOS) my gophish server successfully. My config.json is below.
I created some emails templates, user groups (only my email for the moment), and I am now trying to start a campaign and collect informations. When I receive the emails (as spam for now) and click on it, or download attachments or click on the phishing URL, nothing happens on my gophish campaign, and email is still not marked as read, or linked clicked etc.

I think I have configured correctly my server to be exposed on port 80. I have also tried with a specific DN (that I own, and created a redirection on route53), but no success neither. Note that I have the following entry in my /etc/hosts: 
`81.***.***.9    gophish.le******ce.fr`

What are you expecting to see happen? :

I would like to receive statistics.

config.json :

```html
{
        "admin_server": {
                "listen_url": "0.0.0.0:3333",
                "use_tls": true,
                "cert_path": "le*****ce.crt",
                "key_path": "le*****ce.key",
                "trusted_origins": ["81.***.***.9/32"]
        },
        "phish_server": {
                "listen_url": "0.0.0.0:80",
                "use_tls": true,
                "cert_path": "le*****ce.crt",
                "key_path": "le*****ce.key"
        },
        "db_name": "sqlite3",
        "db_path": "gophish.db",
        "migrations_prefix": "db/db_",
        "contact_address": "",
        "logging": {
                "filename": "",
                "level": ""
        }
}
```

My HTML mail template is the following:
```
<html>
<head>
	<title></title>
</head>
<body>
<p>Hello {{.FirstName}},</p>

<p>Blablalbalbalbabla some random text.</p>

<p>{{.Tracker}}</p>
</body>
</html>

```

And I tried the following URLs for my campaign:
- https://gophish.le******ce.fr, where this domain is owned by me, and responds to `telnet gophish.le******ce.fr 80`, but https://gophish.le******ce.fr/ does not respond if I browse it (however http://gophish.le*******ce.fr/ does respond to a page with a `Web Station has been enabled. To finish setting up your website, please see the "Web Service" section of DSM Help.` written
- http://gophish.le******ce.fr,
- http://0.0.0.0:80/
- http://81.***.***.9:80/ where the IP is the public IP of my machine (on which gophish is running).

What am I missing? Is it because I try to run this on my local machine? 

Thank you very much!
Jules

**Comment by mori1209** [comment]
I have the same problem

**Comment by JulesClaussen** [comment]
For those struggling like me, I found a "workaround".
I'm not sure how I solved the problem, but I believe it's not the best way to implement gophish in local on my machine.
I started up an Ubuntu machine on AWS, and configured it the same way (don't forget to open ports 3333 for gophish to be available), with same certificates, and same config.json, and somehow it worked.

I suspect port 3333 is not publicly available on MacOS by default, hence the app not being available at the HTTPS URL I wanted. Once this is solved, the tracker indeed tracks, because when the users clicks on the URL it redirects towards my custom https URL which is available and send the info back to gophish server.

If I may make a suggestion, I believe this could be a bit clearer in the documentation: the tracker **needs** the app to be publicly available (not on http://0.0.0.0:443/ but on a more classic URL) to work! Basically telnet on port 443 for the domain you host the app must be reachable.

Thank you for the tool anyway! It's quite useful and the UI is quite clear.

**Comment by edermi** [comment]
> I believe this could be a bit clearer in the documentation: the tracker needs the app to be publicly available

Of course stuff needs to be publicly reachable, how is it otherwise supposed to work? I don't think the docs need more clarification on this one. At some point Gophish assumes that you know what you're doing, this includes publishing the service on the internet.
